x-pack/plugins/apm/public/components/alerting/rule_types/error_count_rule_type/index.tsx

```diff
@@ -119,7 +123,7 @@ export function ErrorCountRuleType(props: Props) {
       params.serviceName,
       params.errorGroupingKey,
       params.groupBy,
-      params.kqlFilter,
+      params.searchConfiguration,
```

Comment:
This could be problematic. Before `kqlFilter` was a string, so the identity only changes if the string changes - and thus the preview request will only be made when something the search query changes. Now, the preview request will be made every time `params.searchConfiguration` is defined (happens on every render afaict).

This will fix the problem:

```suggestion
      params.searchConfiguration.query,
      params.searchConfiguration.kuery,
```

Reply:
I've just seen that `query` can be an object, in which case the above won't work

Reply:
I am not sure I follow. I checked the preview request, it is made only when search query changes. Could you give example of other instances when preview request is made?

Reply:
This is a general problem when using non-primitives as dependencies. There are various blog posts on the topic:

- https://haybeecodes.hashnode.dev/having-objects-as-dependencies-in-react-useeffect-hook#heading-problem-with-having-objects-as-dependencies
- https://deepscan.io/docs/rules/react-useless-dependency-of-hook